### PR TITLE
Remove currentDir code from collect assets

### PR DESCRIPTION
**Pull request by ruflin**
The goal is to remove all directory changes in the generator code. This should make the code more predictable and makes it possible to run the generation in parallel.

### CODE REVIEW

File: dev/generator/main.go [hunk 1]
```diff
@@ -113,23 +113,26 @@ func buildPackage(packagesBasePath string, p util.Package) error {
...
-	}
-	defer os.Chdir(currentPath)
 
 	// Checks if the package is valid
 	err = p.Validate()
 	if err != nil {
 		return fmt.Errorf("Invalid package: %s: %s", p.GetPath(), err)
 	}
 
+	p.BasePath = currentPath + "/" + packagesBasePath + "/" + p.GetPath()
```
Comment (mtojek) [inline dev/generator/main.go]: nit: path.Join

File: dev/generator/main.go [hunk 2]
```diff
@@ -113,23 +113,26 @@ func buildPackage(packagesBasePath string, p util.Package) error {
...
+
 	err = p.LoadAssets(p.GetPath())
 	if err != nil {
 		return err
 	}
 
+	err = os.Chdir(packagesBasePath)
```
Comment (mtojek) [inline dev/generator/main.go]: Not blocking, but I'm not convinced if the app should change directory.

Comment (ruflin) [reply dev/generator/main.go]: This is exactly what I'm trying to get rid off. Following up with more PRs.

File: dev/generator/main.go (hunk 1 above)
Comment (ruflin) [reply dev/generator/main.go]: Will do a refactoring for all the path.Join at once, more to come.

**Pull request merged**